plugins/modules/ec2_key.py

```diff
@@ -248,7 +265,22 @@ def _create_key_pair(ec2_client, name, tag_spec, key_type):
     return key
 
 
-def create_new_key_pair(ec2_client, name, key_material, key_type, tags, check_mode):
+def _write_private_key(key_data, file_name):
+    """
+    Write the private key data to the specified file, and remove 'private_key'
+    from the ouput. This ensures we don't expose the key data in logs or task output.
+    """
+    try:
+        with open(file_name, "w") as f:
```

Comment:
The default permissions for new files might be unsafe. Using `os.open()` is probably better, as it allows you to explicitly set the file's mode:
```
            file = os.open(file_name, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, 0o600)
            os.write(file, key_data["private_key"].encode("utf-8"))
            os.close(file)
```
I'm not really familiar with `os.open` though, I've copied this excerpt from https://github.com/ansible-collections/community.crypto/blob/main/plugins/module_utils/io.py#L56.